src/bind.rs

```diff
@@ -273,54 +258,45 @@ where
         let connect = self.transport_registry
             .new_connect(client_ctx.as_ref(), transport::Connect::new(addr, tls));
 
-
-        let log = ::logging::Client::proxy(self.ctx, addr)
-            .with_protocol(protocol.clone());
-        let client = transparency::Client::new(
-            protocol,
-            connect,
-            log.executor(),
-        );
-
-        let sensors = self.sensors.http(
-            client,
-            &client_ctx
-        );
-
-        // Rewrite the HTTP/1 URI, if the authorities in the Host header
-        // and request URI are not in agreement, or are not present.
-        let normalize_uri = NormalizeUri::new(sensors, protocol.was_absolute_form());
-        let upgrade_orig_proto = orig_proto::Upgrade::new(normalize_uri, protocol.is_http2());
-
-        // Automatically perform reconnects if the connection fails.
-        //
-        // TODO: Add some sort of backoff logic.
-        Reconnect::new(upgrade_orig_proto)
-    }
-
-    /// Binds the endpoint stack used to construct a bound service.
+        // TODO: Add some sort of backoff logic between reconnects.
+        self.sensors.http(
```

Comment:
Nit/TIOLI: Personally, I find this much harder to read than the way it was done previously --- I think something like
```rust
let client = transparency::Client::new(protocol, connect, log.executor());
let reconnect = Reconnect::new(client_ctx.clone(), client);
self.sensors.http(client_ctx.clone(), reconnect)
```
This doesn't require me to track the parentheses in my head, and is actually fewer lines...

Reply:
i'm trying to allude to how this might evolve to a pattern more like the following:

``
sensors
    .and_then(reconnect)
    .and_then(client)
    .make_client(&client_ctx)
```

This will change a lot in the near future.